src/components/structures/GroupView.js

```diff
@@ -807,6 +810,11 @@ export default class GroupView extends React.Component {
         showGroupAddRoomDialog(this.props.groupId);
     };
 
+    _dismissUpgradeNotice = () => {
```

Comment:
we should swallow exceptions thrown from `setItem`

e.g. see note in https://developer.mozilla.org/en-US/docs/Web/API/Storage/setItem about private safari

Reply:
We do this almost nowhere else, this should be a platform/maintenance task to abstract LocalStorage away behind something that doesn't explode and handles errors more gracefully rather than try-catching in every call.

The note also points primarily to a browser we treat under `experimental` in our support policy

> Particularly, in Mobile Safari (since iOS 5) it always throws when the user enters private mode.

Reply:
Fair enough about Mobile Safari.

I think the actual handling behaviour needs to differ on a case by case basis, so I'm not sure an abstraction layer would help. For example, in this case, it'd be okay to swallow, but in other cases, it's fairly fatal and we'd want to tell the user. And waiting for an abstraction layer to come along and rescue the situation puts the great in the way of the good..

Reply:
I think in all cases the user should be informed, given that the action they just performed didn't take effect fully.

Reply:
also it exploding is indicative of more explosions being around the corner, given that local storage getting out of sync can be catastrophic for us, that is definitely worth shouting about always.

Reply:
https://github.com/vector-im/element-web/issues/18423